docs/design/design_v1.8.md

```diff
@@ -0,0 +1,437 @@
+
+- Label  the master with `node-role.kubernetes.io/master=""` 
+- Taints  the master with `node-role.kubernetes.io/master:NoSchedule`
+
+Please note that 
+
+1. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+2. Mark master phase can be invoked individually with the `kubeadm phase mark-master` command.
+
+### Configure TLS-Bootstrapping for node joining
+
+Kubeadm uses [Authenticating with Bootstrap Tokens](https://kubernetes.io/docs/admin/bootstrap-tokens/) for joining new nodes to an existing cluster.
+
+`kubeadm init`  ensures that everything is properly configured for this process, and this includes following steps as well as setting API Server and controller flags as already described in previous paragraphs.
+
+#### Save token in a secret for later reference
+
+`kubeadm init`  create a first bootstrap token, either generated automatically or provided by the user with the `--token` flag; as documented in bootstrap token specification, token should be saved as secrets with name `bootstrap-token-<token-id>` under `kube-system` namespace.
+
+Please note that 
+
+1. The token will be used to validate temporary user during TLS bootstrap process; those users will be member of  `system:bootstrappers:kubeadm:default-node-token` group (formerly `system:bootstrappers` in v1.7)
+2. Starting from 1.8 token has a limited validity, default 24Hours (that can be changed with `—token-ttl` flag)
+3. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+4. Additional tokens can be created/managed with the `kubeadm token` command.
+
+#### Allow joining nodes to call CSR API
+
+kubeadm ensure that users in  `system:bootstrappers:kubeadm:default-node-token` group (formerly `system:bootstrappers` in v1.7) are able to access the certificate signing API.
+
+This is implemented by creating a ClusterRoleBinding named `kubeadm:kubelet-bootstrap` between the  group above and the default RBAC role `system:node-bootstrapper`.
+
+Please note that 
+
+1. In case of `kubeadm`  executed in the `--dry-run` mode, this phases targets a "fake API server".
+2. This phase can be invoked individually with the `kubeadm phase bootstrap-token node allow-post-csrs` command.
+
+#### **Setup auto approval for new bootstrap tokens**
+
+kubeadm ensure that bootstrap token request will be automatically approved by the csrapprover controller.
+
+This is implemented by creating ClusterRoleBinding named `kubeadm:node-autoapprove-bootstrap` between the  `system:bootstrappers:kubeadm:default-node-token` group (formerly `system:bootstrappers` in v1.7) and the default role `system:certificates.k8s.io:certificatesigningrequests:nodeclient`.
+
+The role `system:certificates.k8s.io:certificatesigningrequests:nodeclient` should be created as well, granting POST permission to `/apis/certificates.k8s.io/certificatesigningrequests/nodeclient`.
```

Comment:
Mention that this ClusterRole is automatically bootstrapped by the API server in v1.8
Possibly link to docs telling about auto-bootstrapping of RBAC rules